### PR TITLE
Dependencies updates

### PR DESCRIPTION
**Pull request by ljacomet**
* Bump to latest versions
* Fix after API changes
* Clean up properties in build files

### CODE REVIEW

**Comment by anthonydahanne** [comment]
LGTM

File: build.gradle [hunk 1]
```diff
@@ -17,13 +17,26 @@ import scripts.*
...
+  // Third parties
   offheapVersion = '2.2.2'
-  managementVersion = '5.0.0.beta'
   statisticVersion = '1.1.0'
   jcacheVersion = '1.0.0'
   slf4jVersion = '1.7.7'
   sizeofVersion = '0.3.0'
 
+  // Clustered
+  terracottaPlatformVersion = '5.0.3.beta'
```
Comment (anthonydahanne) [inline build.gradle]: 5.0.3.beta2 now

**Comment by anthonydahanne** [comment]
replaced by PR #1323

File: clustered/client/src/test/java/org/ehcache/clustered/client/internal/UnitTestConnectionService.java [hunk 2]
```diff
@@ -331,7 +332,11 @@ public Connection connect(URI uri, Properties properties) throws ConnectionExcep
       throw new IllegalArgumentException("No server available for " + uri);
     }
 
-    Connection connection = serverDescriptor.server.connectNewClient();
+    String name = properties.getProperty(ConnectionPropertyNames.CONNECTION_NAME);
+    if(name == null) {
```
Comment (cljohnso) [inline clustered/client/src/test/java/org/ehcache/clustered/client/internal/UnitTestConnectionService.java]: If you want to get picky, you should have a space between `if` and `(`.

**Pull request closed without merging**